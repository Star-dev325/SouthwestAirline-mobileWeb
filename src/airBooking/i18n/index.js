import airBookingText from 'src/airBooking/i18n/airBookingText';
import confirmFooterText from 'src/airBooking/i18n/confirmFooterText';
import lowFareCalendarText from 'src/airBooking/i18n/lowFareCalendarText';
import popupText from 'src/airBooking/i18n/popupText';
import splitPayText from 'src/airBooking/i18n/splitPayText';
import confirmPassportRequiredText from 'src/airBooking/i18n/confirmPassportRequiredText';

const airBooking = {
  ...airBookingText,
  ...confirmFooterText,
  ...confirmPassportRequiredText,
  ...lowFareCalendarText,
  ...popupText,
  ...splitPayText
};

export default airBooking;
