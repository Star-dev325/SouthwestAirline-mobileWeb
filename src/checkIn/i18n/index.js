import addtionalPassportInfoFormText from 'src/checkIn/i18n/additionalPassportInfoFormText';
import checkInErrorsText from 'src/checkIn/i18n/checkInErrorsText';
import checkInText from 'src/checkIn/i18n/checkInText';
import chooseMobileBoardingPassesText from 'src/checkIn/i18n/chooseMobileBoardingPassesText';
import hazmatText from 'src/checkIn/i18n/hazmatText';
import mobileBoardingPassText from 'src/checkIn/i18n/mobileBoardingPassText';
import permanentResidentCardText from 'src/checkIn/i18n/permanentResidentCardText';
import trackCheckBagsText from 'src/checkIn/i18n/trackCheckBagsText';
import visaText from 'src/checkIn/i18n/visaText';

const checkIn = {
  ...addtionalPassportInfoFormText,
  ...checkInErrorsText,
  ...checkInText,
  ...chooseMobileBoardingPassesText,
  ...hazmatText,
  ...mobileBoardingPassText,
  ...permanentResidentCardText,
  ...trackCheckBagsText,
  ...visaText
};

export default checkIn;
