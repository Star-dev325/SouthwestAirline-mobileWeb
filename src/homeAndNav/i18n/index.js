import emailEnrollText from 'src/homeAndNav/i18n/emailEnrollText';
import homeNavText from 'src/homeAndNav/i18n/homeNavText';
import inFlightText from 'src/homeAndNav/i18n/inFlightText';
import menuListText from 'src/homeAndNav/i18n/menuListText';
import navDrawerText from 'src/homeAndNav/i18n/navDrawerText';
import offersText from 'src/homeAndNav/i18n/offersText';

const homeAndNavText = {
  ...emailEnrollText,
  ...homeNavText,
  ...inFlightText,
  ...menuListText,
  ...navDrawerText,
  ...offersText
};

export default homeAndNavText;
