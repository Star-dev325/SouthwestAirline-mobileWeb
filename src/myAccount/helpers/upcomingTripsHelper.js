import _ from 'lodash';

export const isCompanionEligible = (flights) => _.every(flights, '_links.bookCompanion.href');

export const getOriginDestinationTitle = (reservation) => {
  const outboundSegments = _.get(reservation, 'itinerary.originationDestinations[0].segments');
  const { originationAirportCode } = _.head(outboundSegments);
  const { destinationAirportCode } = _.last(outboundSegments);

  return `${originationAirportCode} - ${destinationAirportCode}`;
};

export const getUserInfoRequest = (userInfo) => ({
  isLoggedIn: true,
  companionInfo: {
    companionFullName: _.get(userInfo, 'companionFullName'),
    companionName: _.get(userInfo, 'companionName')
  }
});

export const getRetrieveReservationInfoFromTrip = (trip, userInfo) => {
  const href = _.get(trip, '_links.viewReservationViewPage.href');

  if (_.isEmpty(href)) {
    return {};
  }

  const recordLocator = href.slice(href.lastIndexOf('/') + 1);
  const query = _.get(trip, '_links.viewReservationViewPage.query');

  return _.merge(
    {
      recordLocator,
      firstName: _.get(query, 'first-name'),
      lastName: _.get(query, 'last-name')
    },
    !_.isEmpty(userInfo) ? getUserInfoRequest(userInfo) : {}
  );
};

export const getCarRetrieveReservationInfoFromTrip = (trip) => {
  const { href, query } = _.get(trip, '_links.carReservationDetails');

  // TODO: we split confirmationNumber by ourself, we should make a new action accept href
  const confirmationNumber = href.slice(href.lastIndexOf('/') + 1);

  return {
    confirmationNumber,
    firstName: query['first-name'],
    lastName: query['last-name'],
    pickupDate: query['pickup-date']
  };
};
