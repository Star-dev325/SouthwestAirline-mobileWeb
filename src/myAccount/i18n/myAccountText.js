export default {
  MY_ACCOUNT__A_LIST: 'A-List',
  MY_ACCOUNT__A_LIST_PREFERRED: 'A-List Preferred',
  MY_ACCOUNT__CANCELLED_FLIGHT_STATUS__CANCELLED: 'CANCELLED',
  MY_ACCOUNT__CLIMBING_TOWARDS: 'Climbing towards ',
  MY_ACCOUNT__COMPANION_PASS: 'Companion Pass',
  MY_ACCOUNT__CONFIRMATION_NUMBER: 'Confirmation #',
  MY_ACCOUNT__CONFIRMATION_NUMBER_SUBTITLE__CONFIRMATION_NUMBER: 'Confirmation # %s',
  MY_ACCOUNT__CONGRATULATIONS: 'Congratulations!',
  MY_ACCOUNT__EARN_YOUR_NEXT: 'Earn your next ',
  MY_ACCOUNT__ENROLL_FOR_RAPID_REWARDS__CONFIRMATION_MESSAGE: "You're successfully enrolled in Rapid Rewards.",
  MY_ACCOUNT__ENROLL_FOR_RAPID_REWARDS__CONFIRMATION_TITLE: 'Confirmation',
  MY_ACCOUNT__ENROLL_FOR_RAPID_REWARDS__EMAIL_OPT_IN: 'Receive promotional email from Southwest',
  MY_ACCOUNT__ENROLL_FOR_RAPID_REWARDS__SUBSCRIPTION_DETAILS: 'Subscription details',
  MY_ACCOUNT__ENROLL_IN_RAPID_REWARDS: 'Enroll in Rapid Rewards',
  MY_ACCOUNT__FLIGHT_CARD__CHECK_PRICE: 'Check price',
  MY_ACCOUNT__FLIGHT_CARD__REBOOK_IT: 'Rebook it',
  MY_ACCOUNT__FLIGHT_CARD__TO: 'to',
  MY_ACCOUNT__FLIGHT_CREDITS: 'Flight Credits',
  MY_ACCOUNT__HEADER__RR_NUMBER: 'RR# %s',
  MY_ACCOUNT__MAINTAIN_YOUR: 'Maintain your ',
  MY_ACCOUNT__MY: 'My',
  MY_ACCOUNT__MY_PROMO_CODES: 'My Promo Codes',
  MY_ACCOUNT__MY_TRIPS: 'My Trips',
  MY_ACCOUNT__MY_TRIPS_PANEL__BOOK_A_TRIP_MESSAGE: 'Book a trip to any one of our over 90 destinations today.',
  MY_ACCOUNT__MY_TRIPS_PANEL__FIND_A_TRIP: 'Find a Trip',
  MY_ACCOUNT__MY_TRIPS_PANEL__LETS_GO: "Let's go",
  MY_ACCOUNT__MY_TRIPS_PANEL__MY_TRIPS_PANEL_HEADING: 'Trips',
  MY_ACCOUNT__MY_TRIPS_PANEL__PAST_FLIGHTS_LABEL: 'Past Flights',
  MY_ACCOUNT__MY_TRIPS_PANEL__SAVED_LABEL: 'Saved Flights',
  MY_ACCOUNT__MY_TRIPS_PANEL__UP_NEXT: 'Up Next',
  MY_ACCOUNT__NO_PROMO_CODE_FOUND_MESSAGE: 'There is no Promo Code associated with your account.',
  MY_ACCOUNT__PAGE_HEADER_TITLE: 'My Account',
  MY_ACCOUNT__PAX_SUB_TITLE__PAX_TYPE_PASSENGER: 'Passenger',
  MY_ACCOUNT__PENDING_STATUS_TIER_TEXT:
    'Beginning January 1, 2022 due to End of Year processing your tier and Companion Pass earning progress will not be displayed. This does not affect your current status. Thank you for your patience.',
  MY_ACCOUNT__PENDING_TIER_STATUS_TEXT:
    'Your tier status is pending due to an end of year processing. This does not affect your status, only the information displayed.',
  MY_ACCOUNT__PROMOTIONS_PAGE__ELIGIBLE_PROMOTIONS: 'Eligible promotion(s)',
  MY_ACCOUNT__PROMOTIONS_PAGE__NO_PROMOTIONS_AVAILABLE: '0 exclusive promotions available',
  MY_ACCOUNT__PROMOTIONS_PAGE__REGISTERED_PROMOTIONS: 'Registered promotion(s)',
  MY_ACCOUNT__PROMOTION_CARD__PROMOTION_DETAILS: 'Promotion details',
  MY_ACCOUNT__PROMOTION_CARD__PROMOTION_REGISTER: 'Register',
  MY_ACCOUNT__PROMOTION_DETAILS_PAGE__PROMOTION_DETAILS: 'Promotion details',
  MY_ACCOUNT__PROMOTION_DETAILS_PAGE__PROMOTION_REGISTER: 'Register',
  MY_ACCOUNT__PROMOTION_YOURE_NOW_REGISTERED: "You're now registered for this promotion.",
  MY_ACCOUNT__PROMO_CODES: 'Promo Codes',
  MY_ACCOUNT__PROMO_CODES_CONSTANT__MAIN_BODY_TEXT:
    'WCM Controlled FPO placeholder for any possible verbiage on promotions, determined by the business.',
  MY_ACCOUNT__PROMO_CODES_CONSTANT__VIEW_PROMO_CODES: 'View Promo Codes',
  MY_ACCOUNT__PTS_GROUP__EARNED_THROUGH: 'Earned through',
  MY_ACCOUNT__PTS_GROUP__PTS: 'PTS',
  MY_ACCOUNT__PTS_GROUP__VIEW_BENEFITS: 'View benefits',
  MY_ACCOUNT__RAPID_REWARDS_ENROLLMENT_FORM: 'RAPID_REWARDS_ENROLLMENT_FORM',
  MY_ACCOUNT__RAPID_REWARDS_MEMBER: 'Rapid Rewards Member',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__CLIMBING: 'CLIMBING',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANIONPASS: 'Companion Pass',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANIONPASS_PRESTRING_MAP__CLIMBING: 'Climbing towards',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANIONPASS_PRESTRING_MAP__CONGRATULATIONS: "You've earned your",
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANIONPASS_PRESTRING_MAP__MAINTAIN: 'Earn your next',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_ALL_MEMBERS_TEXT: '',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_CHASE_MEMBERS_TEXT: '',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_MORE_DETAILS_TITLE: 'More Details',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_READ_MORE: 'Read more.',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_QUALIFYING_POINTS_FOOTNOTE:
    'Companion qualifying points/flights only.',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__CONGRATULATIONS: 'CONGRATULATIONS',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__DOT: '.',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__LABEL_MAP__A_LIST: 'A-List',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__LABEL_MAP__A_LIST_PREFERRED: 'A-List Preferred',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__LABEL_MAP__NON_ELITE: 'Rapid Rewards Member',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__MAINTAIN: 'MAINTAIN',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__NON_ELITE: 'NON_ELITE',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__STATUS_MAP__A_LIST: 'A-List',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__STATUS_MAP__A_LIST_PREFERRED: 'A-List Preferred',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__TIER_PRESTRING_MAP__CLIMBING: 'Climbing towards',
  MY_ACCOUNT__RAPID_REWARDS_PAGE__TIER_PRESTRING_MAP__CONGRATULATIONS: "You're",
  MY_ACCOUNT__RAPID_REWARDS_PAGE__TIER_PRESTRING_MAP__MAINTAIN: 'Maintain your',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_ENROLLED__CALL_TO_ADD_CHANGE_COMPANION: 'Call to add or change Companion',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_ENROLLED__DESIGNATE_ONE_LUCKY_PERSON:
    'Designate one lucky person of your choice to fly with you anytime for a year.',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_ENROLLED__EXCLUSIVE_PROMOTIONS: 'exclusive promotions available',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_ENROLLED__TEL: 'tel:+18004359792',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_ENROLLED__TIER_STATUS_FOOTNOTE: '*Tier qualifying points/flights only.',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_ENROLLED__TIER_STATUS_RESET: 'Resets annually on Jan 1.',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_NOT_ENROLLED__DETAILS_LINK: 'Get details',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_NOT_ENROLLED__ENROLL: 'Enroll',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_NOT_ENROLLED__MESSAGE:
    'Become a Rapid Rewards member today and earn points for every dollar you spend with Southwest and our partners.',
  MY_ACCOUNT__RAPID_REWARDS_PANEL_NOT_ENROLLED__TITLE: 'Reward Yourself',
  MY_ACCOUNT__REGISTERED_MARK__REGISTERED: 'Registered',
  MY_ACCOUNT__STATUS: ' status',
  MY_ACCOUNT__TEASER__BOOK_A_TRIP: 'Book a trip',
  MY_ACCOUNT__TIER_BENEFITS__BENEFIT_LIST_LABEL_A_LIST_PERCENT: '25%',
  MY_ACCOUNT__TIER_BENEFITS__BENEFIT_LIST_LABEL_A_LIST_PREFERRED_PERCENT: '100%',
  MY_ACCOUNT__TIER_BENEFITS__BENEFIT_LIST_LABEL_EARNING_BONUS: 'Earning Bonus',
  MY_ACCOUNT__TIER_BENEFITS__DEFAULT_BENEFIT_DEDICATED_SERVICE_LINE: 'Dedicated Customer Service Phone Line',
  MY_ACCOUNT__TIER_STATUS_GROUP__FLIGHTS_STATUS: 'of %s flights',
  MY_ACCOUNT__TIER_STATUS_GROUP__OR: 'or',
  MY_ACCOUNT__TIER_STATUS_GROUP__POINTS_STATUS: 'of %s points',
  MY_ACCOUNT__UNUSED_FUNDS_CONSTANT__LEARN_MORE: 'Learn More',
  MY_ACCOUNT__UNUSED_FUNDS_CONSTANT__MAIN_BODY_TEXT:
    'Starting December 3, 2019 funds from cancelled tickets which included your Rapid Rewards number and cancelled in accordance with the No Show Policy, or tickets with a valid residual balance will be displayed in your account.',
  MY_ACCOUNT__UNUSED_FUNDS_CONSTANT__VIEW_FUNDS: 'View Funds',
  MY_ACCOUNT__YOURE: "You're ",
  MY_ACCOUNT__YOUVE_EARNED_YOUR: "You've earned your "
};
