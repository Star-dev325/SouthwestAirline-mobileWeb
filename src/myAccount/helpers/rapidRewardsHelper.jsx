import React from 'react';
import _ from 'lodash';
import Icon from 'src/shared/components/icon';
import TierStatusGroupLabel from 'src/myAccount/components/tierStatusGroupLabel';
import i18n from '@swa-ui/locale';

const NON_ELITE_VALUE = 'NON_ELITE';
const A_LIST_VALUE = 'A_LIST';
const A_LIST_PREFERRED_VALUE = 'A_LIST_PREFERRED';
const COMPANION_PASS_VALUE = 'COMPANION_PASS';

const tierMap = [
  {
    label: i18n('MY_ACCOUNT__A_LIST'),
    value: A_LIST_VALUE,
    points: 35000,
    flights: 20
  },
  {
    label: i18n('MY_ACCOUNT__A_LIST_PREFERRED'),
    value: A_LIST_PREFERRED_VALUE,
    points: 70000,
    flights: 40
  }
];

const companionPassMap = {
  label: i18n('MY_ACCOUNT__COMPANION_PASS'),
  value: COMPANION_PASS_VALUE,
  points: 110000,
  flights: 100
};

export const getTierStatusByTierInfo = (tierInfo) => {
  const status = {
    current: {
      label: i18n('MY_ACCOUNT__RAPID_REWARDS_MEMBER'),
      value: NON_ELITE_VALUE
    },
    next: tierMap[0]
  };

  _.forEach(tierMap, (item, index, list) => {
    if (tierInfo && item.value === tierInfo.tier) {
      status.current = _.clone(item);

      let nextIndex = index < list.length - 1 ? index + 1 : index;

      if (tierInfo.tier === A_LIST_VALUE && !_hasAListQualifyingPointsOrFlights(tierInfo)) {
        nextIndex = 0;
      }

      status.next = _.clone(list[nextIndex]);
    }
  });

  return status;
};

export const getTierStatusGroupLabel = (tierInfo) => {
  const { current, next } = getTierStatusByTierInfo(tierInfo);

  return (
    <TierStatusGroupLabel
      next={next}
      current={current}
      isAList={tierInfo.tier === A_LIST_VALUE}
      earnedAListThisYear={_hasAListQualifyingPointsOrFlights(tierInfo)}
      isAListPreferred={tierInfo.tier === A_LIST_PREFERRED_VALUE}
      earnedAListPreferredThisYear={_hasAListPreferredQualifyingPointsOrFlights(tierInfo)}
    />
  );
};

const _hasAListQualifyingPointsOrFlights = (tierInfo) =>
  tierInfo.tierQualifyingPoints >= getTierMapProperty(A_LIST_VALUE, 'points') ||
  tierInfo.tierQualifyingFlights >= getTierMapProperty(A_LIST_VALUE, 'flights');

const _hasAListPreferredQualifyingPointsOrFlights = (tierInfo) =>
  tierInfo.tierQualifyingPoints >= getTierMapProperty(A_LIST_PREFERRED_VALUE, 'points') ||
  tierInfo.tierQualifyingFlights >= getTierMapProperty(A_LIST_PREFERRED_VALUE, 'flights');

export const getCompanionPassStatusGroupLabel = (companionPassInfo) => {
  let preString = '';

  if (_hasCompanionQualifyingPointsOrFlights(companionPassInfo)) {
    preString = i18n('MY_ACCOUNT__YOUVE_EARNED_YOUR');
  } else {
    preString = companionPassInfo.companionPassAchieved
      ? i18n('MY_ACCOUNT__EARN_YOUR_NEXT')
      : i18n('MY_ACCOUNT__CLIMBING_TOWARDS');
  }

  return (
    <span>
      {preString}
      <b> {i18n('MY_ACCOUNT__COMPANION_PASS')}. </b>
    </span>
  );
};

const _hasCompanionQualifyingPointsOrFlights = (companionPassInfo) =>
  companionPassInfo.companionQualifyingPoints >= companionPassMap.points ||
  companionPassInfo.companionQualifyingFlights >= companionPassMap.flights;

export const shouldShowCongratulationForTier = (tierInfo) => {
  const isAListPreferred = tierInfo.tier === A_LIST_PREFERRED_VALUE;

  return isAListPreferred && _hasAListPreferredQualifyingPointsOrFlights(tierInfo);
};

const _shouldShowCongratulationsForCompanionPass = (companionPassInfo) =>
  companionPassInfo && _hasCompanionQualifyingPointsOrFlights(companionPassInfo);

export const getCongratulationForTier = (tierInfo) =>
  (shouldShowCongratulationForTier(tierInfo) ? (
    <span>
      <Icon type="check" />
      {i18n('MY_ACCOUNT__CONGRATULATIONS')}
    </span>
  ) : null);

export const getCongratulationForCompanionPass = (companionPassInfo) =>
  (_shouldShowCongratulationsForCompanionPass(companionPassInfo) ? (
    <span>
      <Icon type="check" />
      {i18n('MY_ACCOUNT__CONGRATULATIONS')}
    </span>
  ) : null);

const getTierMapProperty = (tier, property) => _.chain(tierMap).find({ value: tier }).result(property).value();
