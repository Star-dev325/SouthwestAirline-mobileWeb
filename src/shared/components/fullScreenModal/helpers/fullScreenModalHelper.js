import { history } from 'src/appHistory';
import BrowserObject from 'src/shared/helpers/browserObject';
import { mergeQuery, combineUri, getQueryStringParameterByKey } from 'src/shared/helpers/urlHelper';
import { addEventListenerOnce } from 'src/shared/helpers/eventHelpers';
import Q from 'q';

const { location } = BrowserObject;

export const showFullScreenModal = (id) => {
  if (getQueryStringParameterByKey('_modal') !== id) {
    return Q(history.push(combineUri(location.pathname, mergeQuery({ _modal: id }, location.search))));
  } else {
    return Q();
  }
};

export const hideFullScreenModal = (id) => {
  if (getQueryStringParameterByKey('_modal') === id) {
    const deferred = Q.defer();

    addEventListenerOnce(window, 'popstate', () => {
      deferred.resolve();
    });
    history.goBack();

    return deferred.promise;
  } else {
    return Q();
  }
};

export const hideModalAndUpdateFormField = (id, fieldName, fieldValue, onChange) => {
  hideFullScreenModal(id);
  onChange(fieldName, fieldValue);
};

export const getModalId = () => getQueryStringParameterByKey('_modal');
