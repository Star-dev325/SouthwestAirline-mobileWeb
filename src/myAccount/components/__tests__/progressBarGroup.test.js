jest.mock('@swa-ui/locale');
jest.mock('src/shared/components/fullScreenModal/fullScreenModal', () => (props) => <div>{props.children}</div>);

import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { Provider } from 'react-redux';
import { fireEvent, render } from '@testing-library/react';
import i18n from '@swa-ui/locale';
import createMockStore from 'test/unit/helpers/configureMockStore';
import ProgressBarGroup from 'src/myAccount/components/progressBarGroup';
import * as FullScreenModalHelper from 'src/shared/components/fullScreenModal/helpers/fullScreenModalHelper';

const mockStore = createMockStore();

describe('ProgressBarGroup', () => {
  let hideFullScreenModalFnMock, onClickMock, showFullScreenModalFnMock;

  beforeEach(() => {
    const noop = () => {};

    onClickMock = jest.fn(noop);
    hideFullScreenModalFnMock = jest.fn(noop);
    showFullScreenModalFnMock = jest.fn(noop);

    jest.clearAllMocks();
  });

  it('should render correct component', () => {
    const { container } = createComponent();

    expect(container).toMatchSnapshot();
  });

  describe('shouldCallToAddOrChangeCompanion', () => {
    it('should show the button when shouldCallToAddOrChangeCompanion is true', () => {
      const progressBarGroupComponent = createComponent({ shouldCallToAddOrChangeCompanion: true });

      expect(progressBarGroupComponent.container.querySelector('Button')).not.toBeNull();
    });

    it('should not show the button when shouldCallToAddOrChangeCompanion is false', () => {
      const progressBarGroupComponent = createComponent({ shouldCallToAddOrChangeCompanion: false });

      expect(progressBarGroupComponent.container.querySelector('Button')).toBeNull();
    });
  });

  describe('shouldShowCongratulations', () => {
    it('should show congratulations when showCongratulations is true', () => {
      const progressBarGroupComponent = createComponent({ showCongratulations: true });

      expect(progressBarGroupComponent.container.querySelector('.tier-status-group--label')).not.toBeNull();
    });

    it('should not show congratulations when showCongratulations is false', () => {
      const progressBarGroupComponent = createComponent({ showCongratulations: false });

      expect(progressBarGroupComponent.container.querySelector('.tier-status-group--label')).toBeNull();
    });
  });

  describe('companionPassSection', () => {
    it('should show companion pass section when isCompanion is true', () => {
      const progressBarGroupComponent = createComponent({ isCompanion: true });

      expect(progressBarGroupComponent.container.querySelector('.call-to-companion')).not.toBeNull();
    });

    it('should not show companion pass section when isCompanion is false', () => {
      const progressBarGroupComponent = createComponent({ isCompanion: false });

      expect(progressBarGroupComponent.container.querySelector('.call-to-companion')).toBeNull();
    });

    describe('chase 10k boost message', () => {
      it('should show chaseMembersBoost text', () => {
        const progressBarGroupComponent = createComponent({ isCompanion: true, userAlreadyHasChaseRRVisa: true });

        expect(progressBarGroupComponent.container.querySelector('.call-to-companion--description')).not.toBeNull();
      });

      it('should show chaseMembers boost text when i18n key has value ', () => {
        const i18nOverrideText = {
          MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_CHASE_MEMBERS_TEXT: 'Sample Chase 10k boost text'
        };

        i18n.mockImplementation((i18nKey) => i18nOverrideText[i18nKey]);

        const progressBarGroupComponent = createComponent({ isCompanion: true, userAlreadyHasChaseRRVisa: true });
        const wrapper = progressBarGroupComponent.container.querySelector('.companion-pass-info-container');

        expect(wrapper).not.toBeNull();
        expect(wrapper).toHaveTextContent('Sample Chase 10k boost text');
      });

      it('should not show chaseMembers boost text when value is empty ', () => {
        const i18nOverrideText = {
          MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_CHASE_MEMBERS_TEXT: ''
        };

        i18n.mockImplementation((i18nKey) => i18nOverrideText[i18nKey]);

        const progressBarGroupComponent = createComponent({ isCompanion: true, userAlreadyHasChaseRRVisa: true });
        const wrapper = progressBarGroupComponent.container.querySelector('.companion-pass-info-container');

        expect(wrapper).toBeNull();
      });

      it('should not show chaseMembersBoost text', () => {
        const progressBarGroupComponent = createComponent({ isCompanion: false, userAlreadyHasChaseRRVisa: false });

        expect(progressBarGroupComponent.container.querySelector('.call-to-companion--description')).toBeNull();
      });
    });

    describe('All Member Text', () => {
      it('should show i18n text when key has value ', () => {
        const i18nOverrideText = {
          MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_ALL_MEMBERS_TEXT: 'Sample all members text'
        };

        i18n.mockImplementation((i18nKey) => i18nOverrideText[i18nKey]);

        const progressBarGroupComponent = createComponent({ isCompanion: true });
        const wrapper = progressBarGroupComponent.container.querySelector('.earn-boost');

        expect(wrapper).not.toBeNull();
        expect(wrapper).toHaveTextContent('Sample all members text');
      });

      it('should not show i18n text when the value is empty string', () => {
        const i18nOverrideText = {
          MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_ALL_MEMBERS_TEXT: ''
        };

        i18n.mockImplementation((i18nKey) => i18nOverrideText[i18nKey]);

        const progressBarGroupComponent = createComponent({ isCompanion: true });
        const wrapper = progressBarGroupComponent.container.querySelector('.earn-boost');

        expect(wrapper).toBeNull();
      });

      it('should show modal when clicked on read more link', () => {
        const i18nOverrideText = {
          MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_ALL_MEMBERS_TEXT: 'Sample all members text'
        };

        i18n.mockImplementation((i18nKey) => i18nOverrideText[i18nKey]);

        const progressBarGroupComponent = createComponent({ isCompanion: true });

        jest.spyOn(FullScreenModalHelper, 'showFullScreenModal').mockImplementation(showFullScreenModalFnMock);

        const wrapperBtn = progressBarGroupComponent.container.querySelector('.read-more-link');

        fireEvent.click(wrapperBtn);

        expect(showFullScreenModalFnMock).toHaveBeenCalled();
      });

      it('should close modal when clicked on done button', () => {
        const i18nOverrideText = {
          MY_ACCOUNT__RAPID_REWARDS_PAGE__COMPANION_PASS_POINTS_ALL_MEMBERS_TEXT: 'Sample all members text'
        };

        i18n.mockImplementation((i18nKey) => i18nOverrideText[i18nKey]);

        const progressBarGroupComponent = createComponent({ isCompanion: true });

        jest.spyOn(FullScreenModalHelper, 'showFullScreenModal').mockImplementation(hideFullScreenModalFnMock);

        const wrapperBtn = progressBarGroupComponent.container.querySelector('.read-more-link');

        fireEvent.click(wrapperBtn);

        const moreDetailsDoneBtn = progressBarGroupComponent.container.querySelector('.cancel');

        fireEvent.click(moreDetailsDoneBtn);

        expect(hideFullScreenModalFnMock).toHaveBeenCalled();
      });
    });
  });

  const createComponent = (props = {}) => {
    const state = {
      app: {
        webView: {
          isWebView: false
        }
      },
      router: {
        location: {
          search: '?_modal=COMPANION_PASS_POINTS_READ_MORE_MODAL'
        }
      }
    };
    const defaultProps = {
      onClick: onClickMock,
      isCompanion: true,
      themeColor: 'teal',
      shouldCallToAddOrChangeCompanion: false,
      showCongratulations: false,
      title: {
        preString: '',
        status: '',
        sufString: ''
      },
      pointsDonutProgressBar: {
        percentageComplete: 0,
        pointsRequired: '',
        pointsEarned: ''
      },
      flightsDonutProgressBar: {
        percentageComplete: 0,
        flightsRequired: '0',
        flightsFlown: '0'
      },
      userAlreadyHasChaseRRVisa: false
    };
    const finalProps = {
      ...defaultProps,
      ...props
    };

    return render(
      <Provider store={mockStore(state)}>
        <ProgressBarGroup {...finalProps} />
      </Provider>
    );
  };
});
