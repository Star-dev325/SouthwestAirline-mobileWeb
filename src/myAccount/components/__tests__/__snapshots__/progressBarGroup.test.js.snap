// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarGroup should render correct component 1`] = `
<div>
  <div
    class="progress-bar-group"
  >
    <div
      class="segment"
    >
      <div
        class="progress-bar-group--label"
      >
        <span />
        <strong>
           
          
        </strong>
        <sup>
          ®
        </sup>
      </div>
      <div
        class="progress-bar-group--status"
      >
        <div
          class="progress-bar-group--status-col progress-bar-group--status-col-graph"
        >
          <div
            class="progress-bar-group--status-graph"
          >
            <div
              class="donut-progress-bar"
              style="width: 0px; height: 0px;"
            >
              <svg
                height="0"
                width="0"
              >
                <path
                  d="M -3.5 -3.5 A -5 -5 0 1 0 3.5 -3.5"
                  fill="transparent"
                  stroke="#e6e7e8"
                  stroke-width="10"
                />
                <path
                  class="donut-progress-bar--fill teal"
                  d="M 3.5 -3.5 A -5 -5 0 0 0 3.5 -3.5"
                  fill="transparent"
                  stroke-width="10"
                />
              </svg>
            </div>
            <i
              class="teal icon icon_pts"
            />
          </div>
          <div
            class="progress-bar-group--status-text"
          >
            <label
              class="label"
            />
            <p
              class="desc"
            >
              undefined 
              <sup>
                †
              </sup>
            </p>
          </div>
        </div>
        <div
          class="progress-bar-group--status-col"
        />
        <div
          class="progress-bar-group--status-col progress-bar-group--status-col-graph"
        >
          <div
            class="progress-bar-group--status-graph"
          >
            <div
              class="donut-progress-bar"
              style="width: 0px; height: 0px;"
            >
              <svg
                height="0"
                width="0"
              >
                <path
                  d="M -3.5 -3.5 A -5 -5 0 1 0 3.5 -3.5"
                  fill="transparent"
                  stroke="#e6e7e8"
                  stroke-width="10"
                />
                <path
                  class="donut-progress-bar--fill teal"
                  d="M 3.5 -3.5 A -5 -5 0 0 0 3.5 -3.5"
                  fill="transparent"
                  stroke-width="10"
                />
              </svg>
            </div>
            <i
              class="teal icon icon_airplane-depart"
            />
          </div>
          <div
            class="progress-bar-group--status-text"
          >
            <label
              class="label"
            >
              0
            </label>
            <p
              class="desc"
            >
              undefined 0
              <sup>
                †
              </sup>
            </p>
          </div>
        </div>
      </div>
      <div
        class="call-to-companion"
      >
        <p
          class="call-to-companion--description"
        />
        <div
          class="progress-bar-group--expression-info"
        >
          <sup>
            †
          </sup>
          <p />
          <br />
          <p
            class="ml2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
